File: Websites/SD.IdentitySystem.IPresentation/Models/Authority.cs
using SD.Infrastructure.PresentationBase;

namespace SD.IdentitySystem.IPresentation.Models
{
    /// <summary>
    /// 权限模型
    /// </summary>
    public class Authority : ModelBase
    {
        #region 权限路径 —— string AuthorityPath
        /// <summary>
        /// 权限路径
        /// </summary>
        public string AuthorityPath { get; set; }
        #endregion

        #region 描述 —— string Description
        /// <summary>
        /// 描述
        /// </summary>
        public string Description { get; set; }
        #endregion


        //Others

        #region 信息系统编号 —— string InfoSystemNo
        /// <summary>
        /// 信息系统编号
        /// </summary>
        public string InfoSystemNo { get; set; }
        #endregion

        #region 信息系统名称 —— string InfoSystemName
        /// <summary>
        /// 信息系统名称
        /// </summary>
        public string InfoSystemName { get; set; }
        #endregion

        #region 应用程序类型名称 —— string ApplicationTypeName
        /// <summary>
        /// 应用程序类型名称
        /// </summary>
        public string ApplicationTypeName { get; set; }
        #endregion
    }
}

File: Angular/SD.IdentitySystem.Client/src/models/authority.ts
import {ApplicationType, ModelBase} from "sd-infrastructure";
import {InfoSystem} from "./info-system";

/*权限*/
export interface Authority extends ModelBase {

    /*信息系统编号*/
    infoSystemNo: string;

    /*应用程序类型*/
    applicationType: ApplicationType;

    /*权限路径*/
    authorityPath: string;

    /*描述*/
    description: string | null;

    /*导航属性 - 信息系统*/
    infoSystemInfo: InfoSystem | null;
}
